# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/dnsmasq/allow-raw-hex-string.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

Allow specifing raw, non C, non NUL terminated strings, e.g. for
option 67, bootfile-name on broken vintage Apple OpenFirmware, ...

--- dnsmasq-2.90/src/option.c.vanilla	2024-04-03 13:25:17.415189079 +0200
+++ dnsmasq-2.90/src/option.c	2024-04-03 14:10:41.063242164 +0200
@@ -1633,7 +1633,7 @@
 	}
       /* or names */
       else if (opt_len & (OT_NAME | OT_RFC1035_NAME | OT_CSTRING))
-	is_addr6 = is_addr = is_dec = is_hex = 0;
+	is_addr6 = is_addr = is_dec /*= is_hex */ = 0;
       
       if (found_dig && (opt_len & OT_TIME) && strlen(comma) > 0)
 	{
